test/spmd/test_xla_distributed_checkpoint.py

```diff
@@ -244,6 +247,18 @@ def test_resolve_shard_data(self):
       self.assertTrue(torch.allclose(shard.data, resolved_data))
 
 
+class DistributedCheckpointHelpersTest(DistributedCheckpointTestBase):
+
+  def test_sharded_cpu_state_dict(self):
+    model = self.SimpleLinear().to(xm.xla_device())
+    state_dict = model.state_dict()
+    sharded_cpu_state_dict = _sharded_cpu_state_dict(state_dict)
+    self.assertCountEqual(sharded_cpu_state_dict,
+                          ['fc1.weight', 'fc1.bias', 'fc2.weight', 'fc2.bias'])
+    self.assertTrue(
+        isinstance(sharded_cpu_state_dict['fc1.weight'], _CpuShards))
```

Comment:
Let's also add an assertion that the other tensors are not wrapped in `_CpuShards` but are on the `CPU` device.

```
for name, param in sharded_cpu_state_dict.items():
    if name == 'fc1.weight':
       # Assert isinstance _CpuShards
    else:
      # Assert isinstance torch.Tensor and device is CPU
```